### PR TITLE
show property changes in flip tracking

### LINKED ISSUE #925: Display value modifications for flips
`propertyChanges` is an array of things that changed between buying and selling a flip. Its returned with every flip on the player flips endpoint. There should be some way to access that information on the `/player/{uuid}/flips` page

### PR DESCRIPTION
**Pull request by matthias-luger**
fixes #925

### CODE REVIEW

File: components/FlipTracking/FlipTracking.tsx [hunk 1]
```diff
@@ -143,14 +143,47 @@ export function FlipTracking(props: Props) {
...
+                </div>
+                <p
+                    style={{ cursor: 'pointer' }}
+                    onClick={() => {
+                        let flips = [...trackedFlips]
+                        trackedFlip.showPropertyChanges = !trackedFlip.showPropertyChanges
+                        setTrackedFlips(flips)
+                    }}
+                >
+                    Extra profit calculations: {trackedFlip.showPropertyChanges ? <ArrowDownIcon /> : <ArrowRightIcon />}
```
Comment (Ekwav) [inline components/FlipTracking/FlipTracking.tsx]:
```suggestion
                    Profit changes: {trackedFlip.showPropertyChanges ? <ArrowDownIcon /> : <ArrowRightIcon />}
```

File: components/FlipTracking/FlipTracking.tsx [hunk 2]
```diff
@@ -143,14 +143,47 @@ export function FlipTracking(props: Props) {
...
+                                ))}
+                            </tbody>
+                        </Table>
                     </div>
+                ) : null}
+                <div style={{ display: 'flex', justifyContent: 'end' }}>
+                    <CopyButton
+                        copyValue={isClientSideRendering() ? `${window.location.origin}/player/${router.query.uuid}/flips/${trackedFlip.uId.toString(16)}` : ''}
+                        successMessage={isClientSideRendering() ? <span>{`Copied link to flip!`}</span> : <span />}
+                    />
```
Comment (Ekwav) [inline components/FlipTracking/FlipTracking.tsx]: could the copy button be on the same height as the text if the profit changes are not extended like it was before?  
Currently the button takes an extra line.  
image

Also could you inline the sell time with the finder, and the cherry on top would be if you displayed the relative time like on the auction paage :)

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: components/FlipTracking/FlipTracking.tsx [hunk 3]
```diff
@@ -190,14 +191,59 @@ export function FlipTracking(props: Props) {
...
+                                                        <span style={{ color: 'red', whiteSpace: 'nowrap', marginLeft: '5px' }}>
+                                                            {numberWithThousandsSeperators(change.effect)} Coins
+                                                        </span>
+                                                    )}
+                                                </td>
+                                            </tr>
+                                        ))}
+                                    </tbody>
+                                </Table>
+                            </div>
```
Comment (Ekwav) [inline components/FlipTracking/FlipTracking.tsx:245]: looks like I hoped for except that there is now never more than one property 
image
while the api says there is 
image
also, please use the profit returned from the api as your calculation seems to be wrong

File: components/FlipTracking/FlipTracking.tsx [hunk 4]
```diff
@@ -190,14 +191,59 @@ export function FlipTracking(props: Props) {
...
                                     </span>
                                 }
                                 type={'hover'}
                             />
+                            <span style={{ marginLeft: '15px' }}>
+                                <Tooltip
+                                    type="hover"
+                                    content={
+                                        <span>
+                                            <span className={styles.label}></span> {moment(trackedFlip.sellTime).fromNow()}
```
Comment (Ekwav) [inline components/FlipTracking/FlipTracking.tsx]:
```suggestion
                                            <span className={styles.label}></span>Sold {moment(trackedFlip.sellTime).fromNow()}
```

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: components/FlipTracking/FlipTracking.tsx (hunk 3 above)
Comment (matthias-luger) [reply components/FlipTracking/FlipTracking.tsx:245]: @Ekwav Could you please provide an example?
In my case it shows all the changes: 
image

Comment (matthias-luger) [reply components/FlipTracking/FlipTracking.tsx:245]: And I do use the profit property for that...
image

Comment (Ekwav) [reply components/FlipTracking/FlipTracking.tsx:245]: The issue doesn't appear anymore. 
The auction in the screenshot(s) was http://localhost:3000/player/99ab148c61b146ed9b5d07df7b46984c/flips/9fc596827144

**Review by Ekwav — APPROVED** [review]
LGTM! :+1:

File: components/FlipTracking/FlipTracking.tsx (hunk 3 above)
Comment (Ekwav) [reply components/FlipTracking/FlipTracking.tsx:245]: The issue doesn't appear anymore. 
The auction in the screenshot(s) was http://localhost:3000/player/99ab148c61b146ed9b5d07df7b46984c/flips/9fc596827144

**Pull request merged**